src/app/qgisapp.cpp

```diff
@@ -16046,6 +16046,55 @@ void QgisApp::triggerCrashHandler()
 #endif
 }
 
+void QgisApp::addTabifiedDockWidget( Qt::DockWidgetArea area, QDockWidget *dockWidget, const QStringList &onTopOf )
+{
+  QList< QDockWidget * > dockWidgetsInArea;
+  const auto dockWidgets = findChildren< QDockWidget * >();
+  for ( QDockWidget *w : dockWidgets )
+  {
+    if ( w->isVisible() && dockWidgetArea( w ) == area )
+    {
+      dockWidgetsInArea << w;
+    }
+  }
+
+  addDockWidget( area, dockWidget );  // First add the dock widget, then attempt to tabify
+  if ( dockWidgetsInArea.length() > 0 )
+  {
+    QDockWidget *onTopOfDockWidget = nullptr;
+    if ( !onTopOf.isEmpty() )
+    {
+      QStringListIterator iter( onTopOf );
+      while ( iter.hasNext() )
+      {
+        const QString &objectName = iter.next();
+        for ( QDockWidget *cw : dockWidgetsInArea )
+        {
+          if ( cw->objectName() == objectName )
+          {
+            onTopOfDockWidget = cw;
+            iter.toBack();  // The task is done, exit the iterator
```

Comment:
Not sure if that's the best way to break two iterations (i.e., `iter.toBack()` and `break`).
Please let me know if we should do it differently.

Reply:
I'd just use a standard range based for loop instead of QStringListIterator, and have a "bool found" variable in order to break out of the second loop

Reply:
The iterator was removed in favor of a basic for loop. Thanks for the suggestion.